src/chess/board.cc

```diff
@@ -982,8 +982,32 @@ void ChessBoard::SetFromFen(const std::string& fen, int* rule50_ply,
   string en_passant;
   int rule50_halfmoves;
   int total_moves;
-  fen_str >> board >> who_to_move >> castlings >> en_passant >>
-      rule50_halfmoves >> total_moves;
+  fen_str >> board;
+  if (fen_str.eof()) {
+    who_to_move = "w";
+  } else {
+    fen_str >> who_to_move;
+  }
+  if (fen_str.eof()) {
+    castlings = "-";
```

Comment:
If I understand correctly, this is not what other programs do - if you do another iteration, can you add a comment with any differences you know?

Reply:
Right, the example I linked in https://github.com/LeelaChessZero/lc0/pull/1234#discussion_r414326363 has SF allowing castling with differently positioned "initial" king and rooks whereas currently `ChessBoard::Castlings` assumes the same kingside and queenside rooks for both sides. Keeping with that 960 restriction, I'm not really sure how we would determine if white should get the castling rights or black or neither for https://lichess.org/analysis/r2k3r/8/8/8/8/8/8/1R2K1R1

Reply:
I totally agree, just want to have it documented if it isn't too much trouble. I plan to make it a hard error (if it isn't already - need to ceck) to give a FEN with castling rights for both black and while if the rooks are not in the same column.

Reply:
Added comment and details about outer-most rook.